--- a/chrome/browser/chromeos/login/chrome_restart_request.cc	2013-09-28 13:10:34.000000000 -0400
+++ b/chrome/browser/chromeos/login/chrome_restart_request.cc	2013-10-01 19:00:13.865923791 -0400
@@ -109,6 +109,7 @@
       ::switches::kGpuStartupDialog,
       ::switches::kGpuSandboxAllowSysVShm,
       ::switches::kMultiProfiles,
+      ::switches::kScrollPixels,
       ::switches::kNoSandbox,
       ::switches::kPpapiFlashArgs,
       ::switches::kPpapiFlashInProcess,
--- a/content/browser/renderer_host/render_widget_host_view_gtk.cc	2013-09-28 13:09:51.000000000 -0400
+++ b/content/browser/renderer_host/render_widget_host_view_gtk.cc	2013-10-01 19:03:48.537060065 -0400
@@ -456,6 +456,28 @@
   }
 
   // Return the net up / down (or left / right) distance represented by events
+  // Allow the vertical scroll delta to be overridden from the command line.
+  // This will allow us to test more easily to discover the amount
+  // (either hard coded or computed) that's best.
+  static float GetScrollPixelsPerTick() {
+    static float scroll_pixels = -1;
+    if (scroll_pixels < 0) {
+      // TODO(brettw): Remove the command line switch (crbug.com/63525)
+      scroll_pixels = kDefaultScrollPixelsPerTick;
+      CommandLine* command_line = CommandLine::ForCurrentProcess();
+      std::string scroll_pixels_option =
+          command_line->GetSwitchValueASCII(switches::kScrollPixels);
+      if (!scroll_pixels_option.empty()) {
+        double v;
+        if (base::StringToDouble(scroll_pixels_option, &v))
+          scroll_pixels = static_cast<float>(v);
+      }
+      DCHECK_GT(scroll_pixels, 0);
+    }
+    return scroll_pixels;
+  }
+
+  // Return the net up / down (or left / right) distance represented by events
   // in the  events will be removed from the queue. We only look at the top of
   // queue...any other type of event will cause us not to look farther.
   // If there is a change to the set of modifier keys or scroll axis
@@ -500,7 +522,7 @@
       gdk_event_put(event);
       gdk_event_free(event);
     }
-    return num_clicks * kDefaultScrollPixelsPerTick;
+    return num_clicks * GetScrollPixelsPerTick();
   }
 
   static gboolean OnMouseScrollEvent(GtkWidget* widget,
@@ -524,15 +546,15 @@
     if (event->direction == GDK_SCROLL_UP ||
         event->direction == GDK_SCROLL_DOWN) {
       if (event->direction == GDK_SCROLL_UP)
-        web_event.deltaY = kDefaultScrollPixelsPerTick;
+        web_event.deltaY = GetScrollPixelsPerTick();
       else
-        web_event.deltaY = -kDefaultScrollPixelsPerTick;
+        web_event.deltaY = -GetScrollPixelsPerTick();
       web_event.deltaY += GetPendingScrollDelta(true, event->state);
     } else {
       if (event->direction == GDK_SCROLL_LEFT)
-        web_event.deltaX = kDefaultScrollPixelsPerTick;
+        web_event.deltaX = GetScrollPixelsPerTick();
       else
-        web_event.deltaX = -kDefaultScrollPixelsPerTick;
+        web_event.deltaX = -GetScrollPixelsPerTick();
       web_event.deltaX += GetPendingScrollDelta(false, event->state);
     }
     RenderWidgetHostImpl::From(
--- a/content/public/common/content_switches.cc	2013-09-28 13:09:54.000000000 -0400
+++ b/content/public/common/content_switches.cc	2013-10-01 19:13:29.909224538 -0400
@@ -928,6 +928,11 @@
 const char kChildCleanExit[]                = "child-clean-exit";
 #endif
 
+#if defined(OS_POSIX) && !defined(OS_MACOSX)
+// Specify the amount the trackpad should scroll by.
+const char kScrollPixels[]                  = "scroll-pixels";
+#endif
+
 #if defined(USE_AURA)
 // Forces usage of the test compositor. Needed to run ui tests on bots.
 extern const char kTestCompositor[]         = "test-compositor";
--- a/content/public/common/content_switches.h	2013-09-28 13:09:54.000000000 -0400
+++ b/content/public/common/content_switches.h	2013-10-01 19:13:19.029491015 -0400
@@ -276,6 +276,10 @@
 extern const char kChildCleanExit[];
 #endif
 
+#if defined(OS_POSIX) && !defined(OS_MACOSX)
+extern const char kScrollPixels[];
+#endif
+
 #if defined(USE_AURA)
 CONTENT_EXPORT extern const char kTestCompositor[];
 #endif
